yt_dlp/extractor/yappy.py

```diff
@@ -97,3 +99,32 @@ def _real_extract(self, url):
             'categories': traverse_obj(media_data, ('categories', ..., 'name')) or None,
             'repost_count': int_or_none(media_data.get('sharingCount'))
         }
+
+
+class YappyProfileIE(InfoExtractor):
+    _VALID_URL = r'https?://yappy\.media/profile/(?P<id>\w+)'
+    _TESTS = [{
+        'url': 'https://yappy.media/profile/59a0c8c485e5410b9c43474bf4c6a373',
+        'info_dict': {
+            'id': '59a0c8c485e5410b9c43474bf4c6a373',
+        },
+        'playlist_mincount': 527,
+    }]
+
+    def _real_extract(self, url):
+        profile_id = self._match_id(url)
+
+        def entries(profile_id):
+            for page_num in itertools.count(1):
+                videos = traverse_obj(self._download_json(
+                    f'https://yappy.media/api/video/list/{profile_id}?page={page_num}',
+                    profile_id, f'Downloading profile page {page_num} JSON'),
+                    ('results', lambda _, v: v['uuid']))
+                if not videos:
+                    break
+                for video in videos:
+                    yield self.url_result(
+                        f'https://yappy.media/video/{video["uuid"]}', YappyIE,
+                        video['uuid'], video.get('description'))
```

Comment:
Does each page have a diff number of videos? If so, can merge. If the number is fixed, use pagedlist.

Reply:
> Does each page have a diff number of videos? If so, @bashonly can merge. If the number is fixed, use pagedlist.

All pages (except the last one) contain 15 videos, the last page may have less than 15

Reply:
Then we can use `PagedList`. This allows users to download videos from the middle `-I 30-40` without extra requests.